src/main/java/org/truffleruby/core/encoding/EncodingManager.java

```diff
@@ -57,16 +57,15 @@
 
     public EncodingManager(RubyContext context) {
         this.context = context;
+    }
 
+    public void initializeLocaleEncoding(TruffleNFIPlatform nfi, NativeConfiguration nativeConfiguration) {
         if (TruffleOptions.AOT) {
             // Call setlocale(LC_ALL, "") to ensure the locale is set to the environment's locale
             // rather than the default "C" locale.
             Compiler.command(new Object[]{ "com.oracle.svm.core.posix.PosixUtils.setLocale(String, String)String", "LC_ALL", "" });
         }
```

Comment:
@nirvdrum I moved the `setlocale` call here just before nl_langinfo. I think before that the real locale should not matter.

Reply:
I think it's needed before we make any POSIX I/O calls. I believe we intialize encodings before then, so it should be fine. I just want to make it clear that this code can just keep being pushed down until we call `nl_langinfo`, because it might be too late by then.

Reply:
In this case it's done right after we build the native platform, so it should be fine.